openedx/core/djangoapps/content_libraries/api/blocks.py

```diff
@@ -891,18 +889,14 @@ def publish_component_changes(usage_key: LibraryUsageLocatorV2, user: UserType):
     """
     Publish all pending changes in a single component.
     """
-    content_library = require_permission_for_library_key(
-        usage_key.lib_key,
-        user,
-        CAN_EDIT_THIS_CONTENT_LIBRARY
-    )
+    component = get_component_from_usage_key(usage_key)
+    library_key = usage_key.context_key
+    content_library = ContentLibrary.objects.get_by_key(library_key)  # type: ignore[attr-defined]
     learning_package = content_library.learning_package
-
     assert learning_package
-    component = get_component_from_usage_key(usage_key)
-    drafts_to_publish = authoring_api.get_all_drafts(learning_package.id).filter(
-        entity__key=component.key
-    )
+    # The core publishing API is based on draft objects, so find the draft that corresponds to this component:
+    drafts_to_publish = authoring_api.get_all_drafts(learning_package.id).filter(entity__key=component.key)
+    # Publish the component and update anything that needs to be updated (e.g. search index):
     authoring_api.publish_from_drafts(learning_package.id, draft_qset=drafts_to_publish, published_by=user.id)
     LIBRARY_BLOCK_UPDATED.send_event(
         library_block=LibraryBlockData(
```

Comment:
If we also raise `LIBRARY_CONTAINER_UPDATED` for affected containers here [like we do here](https://github.com/openedx/edx-platform/blob/d7fa5819bf79ed5abb1e1a269032601d1ce9ffd1/openedx/core/djangoapps/content_libraries/api/blocks.py#L292-L301), then it'll handle this case:

1. Edit a Component in a Unit:
    * Component + Unit show "unpublished changes" ✔️ 
2. Publish the Component only.
   * Component no longer shows "unpublished changes" ✔️ 
   * Unit still shows "unpublished changes"

Reply:
Fixed with [c0972ef](https://github.com/openedx/edx-platform/pull/36543/commits/c0972efda37018f4b352cf91beda209f9caec03f).

Reply:
Nice, thanks for identifying and fixing that!